pages/en/lb4/Testing-Your-Extensions.md

```diff
@@ -0,0 +1,303 @@
+recommended to test two items together and other integrations can be
+substituted by test doubles so it becomes apparent where the integration breaks
+(if it does).
+
+#### Controllers
+
+*TBD ... Repositories? Other Services?*
+
+#### Decorators
+
+*TBD ... How / What?*
+
+#### Mixins
+
+As mentioned under unit/mixins section, a Mixin extends a base Class by returning
+an anonymous class. As such to test a Mixin, it should be done by actually
+using the mixin with it's base Class. Since this requires two Classes to work
+together, an integration test will be needed. A Mixin should be tested to ensure
+new / overridden methods exist and work as expected in the new Mixed class. 
+
+**`src/mixins/time.mixin.ts`**
+```ts
+import {Constructor} from '@loopback/context';
+export function TimeMixin<T extends Constructor<any>>(superClass: T) {
+  return class extends superClass {
+    constructor(...args: any[]) {
+      super(...args);
+      if (!this.options) this.options = {};
+
+      if (typeof this.options.timeAsString !== 'boolean') {
+        this.options.timeAsString = false;
+      }
+    }
+
+    time() {
+      if (this.options.timeAsString) {
+        return new Date().toString();
+      }
+      return new Date();
+    }
+  };
+}
+```
+
+**`test/integration/mixins/time.intg.test.ts`**
+```ts
+import {expect} from '@loopback/testlab';
+import {Application} from '@loopback/core';
+import {TimeMixin} from '../../..';
+
+describe('TimeMixin (integration)', () => {
+  it('mixed class has .time()', () => {
+    const myApp = new AppWithTime();
+    expect(typeof myApp.time).to.be.eql('function');
+  });
+
+  it('returns time as string', () => {
+    const myApp = new AppWithLogLevel({
+      timeAsString: true,
+    });
+
+    const time = myApp.time();
+    expect(time).to.be.a.String();
+  });
+
+  it('returns time as Date', () => {
+    const myApp = new AppWithLogLevel();
+
+    const time = myApp.time();
+    expect(time).to.be.a.Date();
+  });
+
+  class AppWithTime extends TimeMixin(Application) {}
+});
+```
+
+#### Providers
+
+*TBD ... How / What?*
+
+#### Repositories
+
+*TBD ... How / What*
+
+### Acceptance Test
+
+An Acceptance test is the ultimate test. This is a comprehensive test, written
+end-to-end that is responsible for covering the user scenario. This acceptance
+test should make use of all extension artifacts such as decorators, mixins,
+providers, repositories, etc. No test doubles should need to be used for an
+Acceptance Test. This is a black box test where you don't know / care about the
+internals of the extensions. You will be thinking like the consumer of the
+extension (using it, and it doing a thing as expected).
+
+Due to the complexity of an acceptance test, there is no example given here.
+Have a look at a [loopback4-example-log-extension](https://github.com/strongloop/loopback4-example-log-extension)
+to understand the extension artifacts and usage. Then you can see the acceptance
```

Comment:
`and usage`
and their usage
doesn't matter too much imo

Reply:
`Then you can see the acceptance ...`
Awkward phrasing. I'd go with something like 'You can see its acceptance test here'